src/coreclr/jit/redundantbranchopts.cpp

```diff
@@ -47,7 +47,29 @@ PhaseStatus Compiler::optRedundantBranches()
             if (block->bbJumpKind == BBJ_COND)
             {
                 madeChanges |= m_compiler->optRedundantRelop(block);
+
+                BasicBlock* bbNext = block->bbNext;
+                BasicBlock* bbJump = block->bbJumpDest;
+
                 madeChanges |= m_compiler->optRedundantBranch(block);
+
+                // It's possible that either bbNext or bbJump were unlinked and it's proven
+                // to be profitable to pay special attention to their successors.
+                if (madeChanges && (bbNext != nullptr) && (bbNext->countOfInEdges() == 0))
```

Comment:
shouldn't `bbNext` always be non-null (since we're at a `BBJ_COND`) and likewise for `bbJump`?